docusaurus/docs/React/custom-code-examples/video-integration-100ms.mdx

```diff
@@ -0,0 +1,881 @@
+
+First, take a look at the `tsx` file. We want it to be a component that shows the `channelName` so we need to get the current `channel` object from `useChannelStateContext()`. For that, we’ll use an import and then fetch it. Let’s first create a small dummy `<h2>` element that will show the channel name if available and `Unknown` otherwise:
+
+```tsx
+import { useChannelStateContext } from 'stream-chat-react';
+
+const MyChannelHeader = ({ channelName }: MyChannelHeaderProps) => {
+  const { channel } = useChannelStateContext();
+
+  return <h2>{channel?.data?.name ?? 'Unknown'}</h2>;
+};
+
+export default MyChannelHeader;
+```
+
+Next, we’ll use the `VideoContext` to import the necessary functions we need and create three helper functions:
+
+1. `onVideoButtonClick`: When the user clicks on the video button there should be different actions called depending on the current state of video from the context
+2. `callArea`: to show the correct button(s) or state indicators depending on the current `connectionState`
+3. `buttonText`: depending on the state of connection we will re-use the button to show different text
+
+Here is the code for the three functions, please add this inside of your `MyChannelHeader` component:
+
+```tsx
+const { connectionState, createCall, joinCall, leaveCall, endCall } = useVideoContext();
+
+const onVideoButtonClick = () => {
+  switch (connectionState) {
+    case ConnectionState.NoCall:
+      createCall();
+      break;
+    case ConnectionState.CallAvailable:
+      joinCall();
+      break;
+    case ConnectionState.InCall:
+      leaveCall();
+      break;
+  }
+};
+
+const callArea = (): ReactNode => {
+  switch (connectionState) {
+    case ConnectionState.NoCall:
+    case ConnectionState.CallAvailable:
+      return (
+        <button className='call-button start-call-button' onClick={onVideoButtonClick}>
+          {buttonText()}
+        </button>
+      );
+    case ConnectionState.Connecting:
+    case ConnectionState.Disconnecting:
+      return <p className='call-button'>{buttonText()}</p>;
+    case ConnectionState.InCall:
+      return (
+        <>
+          <button className='call-button leave-call-button' onClick={onVideoButtonClick}>
+            <p>{buttonText()}</p>
+          </button>
+          <button className='call-button end-call-button' onClick={endCall}>
+            <p>End call</p>
+          </button>
+        </>
+      );
+  }
+};
+
+const buttonText = (): string => {
+  switch (connectionState) {
+    case ConnectionState.NoCall:
+      return 'Create call';
+    case ConnectionState.CallAvailable:
+      return 'Join call';
+    case ConnectionState.Connecting:
+      return 'Connecting';
+    case ConnectionState.Disconnecting:
+      return 'Disconnecting';
+    case ConnectionState.InCall:
+      return 'Leave Call';
+  }
+};
```

Comment:
```suggestion
const buttonText = (connectionState: ConnectionState) => {
  switch (connectionState) {
    case ConnectionState.NoCall:
      return 'Create call';
    case ConnectionState.CallAvailable:
      return 'Join call';
    case ConnectionState.Connecting:
      return 'Connecting';
    case ConnectionState.Disconnecting:
      return 'Disconnecting';
    case ConnectionState.InCall:
      return 'Leave Call';
  }
};
```
This function then can live outside the component scope to keep its reference stable.

Reply:
Good catch! Refactored it!